### PR TITLE
feat(contracts): bond validator KRO

### PR DESCRIPTION
**Pull request by seolaoh**
# Description

Bond validator KRO during output submission and challenge creation.

### CODE REVIEW

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 1]
```diff
@@ -77,15 +72,9 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
...
+     * @notice The amount to slash.
      */
-    uint128 public immutable SLASHING_RATE;
-
-    /**
-     * @notice Minimum amount to slash. It should be equal or less than
-     *         ValidatorManager.MIN_ACTIVATE_AMOUNT.
-     */
-    uint128 public immutable MIN_SLASHING_AMOUNT;
+    uint128 public immutable SLASHING_AMOUNT;
```
Comment (0xHansLee) [inline packages/contracts/contracts/L1/AssetManager.sol:77]: Why don't we name this, `BOND_AMOUNT`?

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 2]
```diff
@@ -678,6 +669,29 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
...
+     *         function is only called by the ValidatorManager contract.
+     *
+     * @param validator Address of the validator.
+     */
+    function bondValidatorKro(address validator) external onlyValidatorManager {
+        Vault storage vault = _vaults[validator];
+
+        unchecked {
+            if (vault.asset.validatorKro - vault.asset.validatorKroBonded < SLASHING_AMOUNT)
+                revert InsufficientAsset();
```
Comment (0xHansLee) [inline packages/contracts/contracts/L1/AssetManager.sol:683]: Let's take out this from `unchecked` statement. Underflow can happen.

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 3]
```diff
@@ -678,6 +669,29 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
...
+
+            vault.asset.validatorKroBonded += SLASHING_AMOUNT;
+
+            emit ValidatorKroBonded(
+                validator,
+                SLASHING_AMOUNT,
+                vault.asset.validatorKro - vault.asset.validatorKroBonded
+            );
```
Comment (0xHansLee) [inline packages/contracts/contracts/L1/AssetManager.sol:691]: same here.

File: packages/contracts/contracts/L1/interfaces/IAssetManager.sol [hunk 4]
```diff
@@ -208,14 +208,27 @@ interface IAssetManager {
     );
 
     /**
-     * @notice Reverts when caller is not allowed.
+     * @notice Emitted when validator KRO is bonded during output submission or challenge creation.
+     *
+     * @param validator Address of the validator.
+     * @param amount    The amount of KRO bonded.
+     * @param remainder The remaining amount of validator KRO excluding bonded KRO.
```
Comment (0xHansLee) [inline packages/contracts/contracts/L1/interfaces/IAssetManager.sol:215]: This seems good.

File: packages/contracts/contracts/L1/AssetManager.sol (hunk 1 above)
Comment (sm-stack) [reply packages/contracts/contracts/L1/AssetManager.sol:77]: I also it would be better for code readability

**Review by 0xHansLee — APPROVED** [review]

**Review by sm-stack — APPROVED** [review]
LGTM

**Pull request merged**